core/document/src/layers/folder.rs

```diff
@@ -153,15 +141,19 @@ impl Folder {
 	}
 
 	pub fn layer(&self, id: LayerId) -> Option<&Layer> {
-		let pos = self.layer_ids.iter().position(|x| *x == id)?;
+		let pos = self.position_of_layer(id)?;
```

Comment:
I would probably use `ok()` here and return a result to reduce code duplication

Reply:
Yep, that works.